inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -186,6 +199,8 @@ public synchronized Runnable start(ManagedClientTransport.Listener listener) {
         serverSchedulerPool = server.getScheduledExecutorServicePool();
         serverScheduler = serverSchedulerPool.getObject();
         serverStreamTracerFactories = server.getStreamTracerFactories();
+        assumedMessageSize =
+            (assumedMessageSize == -1) ? server.getAssumedMessageSize() : assumedMessageSize;
```

Comment:
@ejona86 I made this conditional change in response to this [comment](https://github.com/grpc/grpc-java/pull/11406#discussion_r1757048666)

By doing this, assumedMessageSize will take the value from client provided if it was not provided by server.

Reply:
Hi @ejona86 please review this.

Reply:
Hi Shiva,
  You should only have the assumedMessageSize on one side and then have both sides use that.  You shouldn't allow for values to be set separately for client and server side, so remove it from either InProcessChannelBuilder or InProcessServerBuilder.

Reply:
Okay. I removed it from InProcessServerBuilder.